design-documents/service-isolation/inter-service-communication.md

```diff
@@ -0,0 +1,301 @@
+### Inter-service communications
+This documents describes how exactly services will be communicating in a distributed setup.
+#### Service declaration
+Services will be declared in Api modules using interfaces. These modules will be installed on all nodes.
+Example:
+```php
+namespace Magento\ServiceBApi;
+
+interface BManagerInterface
+{
+    public function save(BDataInterface $data): BDataInterface;
+    
+    public function delete(string $id): void;
+}
+```
+or for DTOs
+```php
+namespace Magento\ServiceBApi\Data;
+
+interface BDataInterface
+{
+    public function getId(): string;
+  
+    public function getName(): string;
+}
+```
+The API modules will just contain these declarations and config files (like acl.xml, config.xml) describing module
+without describing actual implementation.
+ 
+Even services that are not written using Magento (or even PHP) will be declared in this way for nodes using Magento.
```

Comment:
What do you mean here, could you elaborate?

Reply:
Not all programming languages support interfaces. If "tomorrow" we need to write a new service on a different stack, how invoker will call it if the invoker?